### PR TITLE
Computation reports for Cadence scripts & transactions

### PR DESCRIPTION
**Pull request by m-Peter**
Completes: 1st milestone of https://github.com/onflow/developer-grants/issues/178
Closes: https://github.com/onflow/flow-emulator/issues/526

## Description

Introduces a new endpoint under `http://localhost:8080/emulator/computationReport`, to display computation reports. For example:

computation-profile

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: emulator/computation_profile.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		MemoryEstimate:  txResult.MemoryEstimate,
+		Code:            code,
+		Arguments:       arguments,
+	}
+	cp.Transactions[txResult.TransactionID.String()] = txReport
+}
+
+// transformIntensities maps a numeric common.ComputationKind value
+// to a human-friendly string value.
+func transformIntensities(
```
Comment (janezpodhostnik) [inline emulator/computation_profile.go:78]: I wish flow-go already offered ComputationKind names.

File: emulator/computation_profile_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+	expectedIntensities := map[string]uint{
+		"FunctionInvocation":     2,
+		"GetAccountContractCode": 1,
+		"GetCode":                1,
+		"GetOrLoadProgram":       3,
+		"GetValue":               221,
+		"ResolveLocation":        1,
+		"Statement":              1,
+	}
```
Comment (janezpodhostnik) [inline emulator/computation_profile_test.go:94]: I would not compare all the values, as the test might become fragile.
I would maybe omit `GetValue` and perhaps `GetOrLoadProgram` as well.

File: emulator/computation_profile_test.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+		"GenerateAccountLocalID": 1,
+		"GenerateUUID":           1,
+		"GetAccountContractCode": 1,
+		"GetCode":                1,
+		"GetOrLoadProgram":       3,
+		"GetValue":               2249,
+		"ResolveLocation":        1,
+		"SetValue":               2473,
+		"Statement":              11,
+		"TransferCompositeValue": 3,
```
Comment (janezpodhostnik) [inline emulator/computation_profile_test.go:160]: I would not check:
- EncodeValue
- GetValue
- SetValue
- AllocateStorageIndex
- GetOrLoadProgram

**Review by janezpodhostnik — APPROVED** [review]
Looks good. Some comments:

- It might be good to link to this function as an example on how to get the computation from the intensities https://github.com/onflow/flow-go/blob/master/fvm/meter/computation_meter.go#L32
- The naming: "profiling" hints to the profiling where you can see results per line of code executed which is a bit misleading. However if I understand correctly you intend to work towards that. If that is the case I'm ok with the name.

File: emulator/computation_profile.go (hunk 1 above)
Comment (m-Peter) [reply emulator/computation_profile.go:78]: Unfortunately, I think this is not possible. The main problem is that `ComputationKind` is defined in Cadence repository and it uses the `string` cmd tool to generate the `String` representation for the values defined there. See: https://github.com/onflow/cadence/blob/master/runtime/common/compositekind.go#L27
Only the `ComputationKind` values defined in Cadence offer a human-readable name: https://github.com/onflow/cadence/blob/master/runtime/common/computationkind_string.go.

File: emulator/computation_profile_test.go (hunk 3 above)
Comment (m-Peter) [reply emulator/computation_profile_test.go]: Good point :+1: 
Updated in https://github.com/onflow/flow-emulator/pull/609/commits/5dc4699492f9e2044349dedb25ef33dbf45eb23d

**Comment by m-Peter** [comment]
@janezpodhostnik I have updated the naming pattern to `computation report` in https://github.com/onflow/flow-emulator/pull/609/commits/7a259eb77a41669fa66cc7e88babd5c54ba61f34, for types / variables / flags etc. I have also linked the relevant flow-go function for getting the computation from intensities.

**Pull request merged**